04/2_print_string_buffer/fixed_string.c

```diff
@@ -0,0 +1,27 @@
+#include <stdio.h>
+#include <string.h>
+
+#include "fixed_string.h"
+
+static int s_is_called;
+static char s_buffer[BUFFER_LENGTH];
+
+int copy_string(const char* src, size_t num)
+{
+    if (!s_is_called) {
+        strncpy(s_buffer, src, num);
+        s_is_called = TRUE;
+    } else {
+        strncat(s_buffer, src, num);
+    }
+
+    s_buffer[BUFFER_LENGTH - 1] = '\0';
+    
+    if (strlen(s_buffer) == BUFFER_LENGTH - 1) {

```

Comment:
when buffer is full. you simply rpint that to printf. and the should add unprinted string into buffer again